application/src/main/java/com/sanction/lightning/authentication/LightningAuthenticator.java

```diff
@@ -0,0 +1,37 @@
+package com.sanction.lightning.authentication;
+
+import com.google.common.base.Optional;
+
+import io.dropwizard.auth.AuthenticationException;
+import io.dropwizard.auth.Authenticator;
+
+import io.dropwizard.auth.basic.BasicCredentials;
+
+import java.util.List;
+
+public class LightningAuthenticator implements Authenticator<BasicCredentials, Key> {
+
+  private final List<Key> allKeys;
+
+  public LightningAuthenticator(List<Key> keys) {
+    this.allKeys = keys;
+  }
+
+  @Override
+  public Optional<Key> authenticate(BasicCredentials credentials) throws AuthenticationException {
+
+    // Construct a key from incoming credentials
+    Key key = new Key(credentials.getUsername(), credentials.getPassword());
+
+    if (key == null) {
```

Comment:
You don't need this `if`, you construct a `new Key` right above it, so it can't be `null`